src/MonitorProfiler/Logging/DebugLogger.cpp

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "DebugLogger.h"
+#include "LogLevelHelper.h"
+
+using namespace std;
+
+STDMETHODIMP DebugLogger::Log(LogLevel level, const string format, va_list args)
+{
+    HRESULT hr = S_OK;
+
+    CHAR szMessage[s_nMaxEntrySize];
```

Comment:
Should we be using Unicode for logging? A lot of the data we are going to be working with will likely be WCHAR. We probably don't want to be doing conversions constantly when logging.

Reply:
I know we chatted about this, but I think I'm going to defer this for now, but try to resolve it soon.

There's a couple of things I need to figure out first, such as:
- While the debug option logger can use `OutputDebugStringW` natively, the formatting functions will not support `WCHAR*` strings. Thus, it would require conversion from `WCHAR*` to `CHAR*` first and then output using `OutputDebugStringA`. Alternatively, the format string could be parsed to find the format specifiers, format each value individually, and recombine before printing. This is what the PAL does, but we haven't taken any PAL implementations yet and I'm hesitant to write this myself.
- In either case (`string` vs `tstring`), mixing string types for the log statements is going to be problematic. For example, the `IfFailLogRet` macros use `__func__` to write the function name out the log; this variable is typically a `char*`. If we were to convert the `Log` methods to use `tstring`, I think we'd expect the variable list to be non-`char` types. So then we get into having to inspect each value in the `va_list`, determining if it is `string`-y or `char`-y, and convert it to the appropriate `string` or `char` type.

A possible solution here is to have a `Log` overload for `string` and for `tstring`; caller chooses which one to use and must use it consistently with the args. Each logger can determine to which size it should be converted before writing it out.

Reply:
I was able to convert to using tstring fairly easily: https://github.com/dotnet/dotnet-monitor/pull/1775

We'll just have to keep in mind that we cannot pass `CHAR` or `CHAR*` args to the formatter.